evm/logic/system.py

```diff
@@ -52,7 +74,7 @@ def create(computation):
 
     call_data = computation.memory.read(start_position, size)
 
-    create_msg_gas = computation.gas_meter.gas_remaining
+    create_msg_gas = max_child_gas_modifier(computation.gas_meter.gas_remaining)
```

Comment:
The last commit changes it to use a class like Call. I don't find it significantly cleaner, but I don't feel strong about either way